engine/runner/src/main/scala/org/enso/runner/MainUtil.scala

```diff
@@ -0,0 +1,179 @@
+/** Remaining utilities for the Main CLI class. */
+private[runner] object MainUtil {
+  final def printPolyglotException(
+    exception: PolyglotException,
+    relativeTo: Option[File]
+  ): Unit = {
+    val fullStack = exception.getPolyglotStackTrace.asScala.toList
+    val dropInitJava = fullStack.reverse
+      .dropWhile(_.getLanguage.getId != LanguageInfo.ID)
+      .reverse
+    val msg: String = HostEnsoUtils.findExceptionMessage(exception)
+    println(s"Execution finished with an error: ${msg}")
+    def printFrame(frame: PolyglotException#StackFrame): Unit = {
+      val langId =
+        if (frame.isHostFrame) "java" else frame.getLanguage.getId
+      val fmtFrame = if (frame.getLanguage.getId == LanguageInfo.ID) {
+        val fName = frame.getRootName
+        val src = Option(frame.getSourceLocation)
+          .map { sourceLoc =>
+            val ident = Option(sourceLoc.getSource.getPath)
+              .map { path =>
+                relativeTo match {
+                  case None => path
+                  case Some(root) =>
+                    val absRoot = root.getAbsoluteFile
+                    if (path.startsWith(absRoot.getAbsolutePath)) {
+                      val rootDir =
+                        if (absRoot.isDirectory) absRoot
+                        else absRoot.getParentFile
+                      rootDir.toPath.relativize(new File(path).toPath).toString
+                    } else {
+                      path
+                    }
+                }
+              }
+              .getOrElse(sourceLoc.getSource.getName)
+            val loc = if (sourceLoc.getStartLine == sourceLoc.getEndLine) {
+              val line  = sourceLoc.getStartLine
+              val start = sourceLoc.getStartColumn
+              val end   = sourceLoc.getEndColumn
+              s"$line:$start-$end"
+            } else {
+              s"${sourceLoc.getStartLine}-${sourceLoc.getEndLine}"
+            }
+            s"$ident:$loc"
+          }
+          .getOrElse("Internal")
+        s"$fName($src)"
+      } else {
+        frame.toString
+      }
+      println(s"        at <$langId> $fmtFrame")
+    }
+    if (exception.isSyntaxError()) {
+      // no stack
+    } else if (dropInitJava.isEmpty) {
+      fullStack.foreach(printFrame)
+    } else {
+      dropInitJava.foreach(printFrame)
+    }
+  }
+
+  /** Handles `--server` CLI option
+    *
+    * @param line     a CLI line
+    * @param logLevel log level to set for the engine runtime
+    */
+  final def runLanguageServer(line: CommandLine, logLevel: Level): Unit = {
+    val maybeConfig = parseServerOptions(line)
+
+    maybeConfig match {
+      case Left(errorMsg) =>
+        System.err.println(errorMsg)
+        Main.exitFail()
+
+      case Right(config) =>
+        LanguageServerApp.run(
+          config,
+          logLevel,
+          line.hasOption(Main.DAEMONIZE_OPTION)
+        )
+        Main.exitSuccess()
+    }
+  }
+
+  final def parseServerOptions(
```

Comment:
Why keep this in Scala when the purpose of this PR is to remove it? It's pretty confusing right now when we should use Scala and when Java for cli parsing.

Reply:
The problem was that I was afraid of `Either` and yes, rewriting that to Java isn't completely easy: 4c22e4e. I hope I got it right...

Reply:
`MainUtil.scala` has been deleted in aad9c11